components/form-builder/app/responses/Pagination.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect } from "react";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { BackArrowIcon, ForwardArrowIcon } from "@components/form-builder/icons";
+import { useTranslation } from "react-i18next";
+
+export const Pagination = ({
+  lastEvaluatedKey,
+  formId,
+  responseDownloadLimit,
+  recordCount,
+}: {
+  lastEvaluatedKey: Record<string, any> | null | undefined;
+  formId: string;
+  responseDownloadLimit: number;
+  recordCount: number;
+}) => {
+  const { t } = useTranslation("form-builder-responses");
+  const router = useRouter();
+
+  // Extract responseId from lastEvaluatedKey object
+  const lastEvaluatedResponseId = lastEvaluatedKey
+    ? lastEvaluatedKey.NAME_OR_CONF.split("#")[1]
+    : "end"; // If lastEvaluatedKey is null, we're on the last page
+
+  // Need statusQuery when building up the prev/next links
+  let statusQuery;
+  if (router.query.params) {
+    [, statusQuery] = router.query.params;
+  }
+
+  // Keep track of the start point for each of our pages.
+  // To help when determining whether we're at the beginning or end,
+  // the first item in the array is always "start" and the last item is always "end"
+  const [pages, setPages] = React.useState<string[]>(["start"]);
+
+  // Update our pages state when the query changes
+  useEffect(() => {
+    setPages(router.query.pages ? String(router.query.pages).split(",") : ["start"]);
```

Comment:
Might be easier to read if we pull pages off into a variable as your using it multiple times.